tests/test_views.py

```diff
@@ -0,0 +1,1050 @@
+            2,
+            url_project=project,
+            value=1
+        )
+        response = getattr(client, request_type)(
+            reverse('project_urls', args=[project.project_slug]))
+
+        assert response.context['project'] == project
+        assert response.context['url_count'] == 3
+        assert response.context['nominator_count'] == 2
+        for key, value in response.context['browse_tup']:
+            assert key == 'com'
+            for each in value:
+                if each[1] is not None:
+                    assert each == ('E', 'http://(com,e')
+
+
+class TestUrlListing():
+
+    @pytest.mark.parametrize('request_type, create_url', [
+        ('get', False),
+        ('get', True),
+        ('post', False),
+        ('post', True)
+    ])
+    def test_status_ok(self, client, request_type, create_url):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory()
+        if create_url:
+            factories.URLFactory(url_project=project, entity=entity)
+        response = getattr(client, request_type)(
+            reverse('url_listing', args=[project.project_slug, entity]))
+
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize('request_type, create_url, expected_template', [
+        ('get', False, 'nomination/url_add.html'),
+        ('post', False, 'nomination/url_add.html'),
+        ('get', True, 'nomination/url_listing.html'),
+        ('post', True, 'nomination/url_listing.html')
+    ])
+    def test_template_used(self, client, request_type, create_url, expected_template):
+        entity = 'www.example.com'
+        project = factories.ProjectFactory()
+        if create_url:
+            factories.URLFactory(url_project=project, entity=entity)
+        response = getattr(client, request_type)(
+            reverse('url_listing', args=[project.project_slug, entity]))
+
+        assert response.templates[0].name == expected_template
+
+    def test_context_url_not_found(self, client):
+        entity = 'http://www.example.com'
+        project = factories.ProjectFactory()
+        response = client.get(reverse('url_listing', args=[project.project_slug, entity]))
+        expected_context = {
+            'project': project,
+            'form': views.URLForm({'url_value': entity}),
+            'url_not_found': True,
+            'metadata_vals': views.get_metadata(project),
+            'form_errors': None,
+            'summary_list': None,
+            'json_data': None,
+            'form_types': None,
+            'institutions': views.get_look_ahead(project),
+            'url_entity': entity,
+        }
+
+        for key in expected_context:
+            assert str(response.context[key]) == str(expected_context[key])
+
+    def test_base_context_values(self, client):
+        """Test the context variables which are similar across different paths.
+
+        These context variables do not change based on the specific path taken
```

Comment:
Unfortunately, yes. Trying to combine them would make the test pretty difficult to read through. And splitting them further apart leads to excessive repetition. However, I'm open to your suggestions.